Gopkg.toml

```diff
@@ -28,3 +28,7 @@ required = [
 [[constraint]]
   name = "github.com/percona/exporter_shared"
   version = "0.7.2"
+
+[[constraint]]
+  name = "github.com/mxschmitt/golang-combinations"
```

Comment:
Hard no. As we discussed, we should keep only actually used combination. We can't afford to generate 2^N combinations only to use a few of them.

Instead, we should iterate over rules and keep unique combinations of channels

Reply:
> we should keep only actually used combination

by actually used combination do you mean the combination of channelIDs used in the rule as one combination and then each individual channel?

for e.g. if there is a rule with 3 channels `/channel_id/1`, `/channel_id/2 `, `/channel_id/3` the receivers should look like this?

```
receivers:
  - name: empty
 
  - name: /channel_id/1
 
  - name: /channel_id/2

  - name: /channel_id/3
 
  - name: /channel_id/1 + /channel_id/2 + /channel_id/3
```

Reply:
Only "empty" (which is special) and "/channel_id/1 + /channel_id/2 + /channel_id/3". Three channels in the middle are not used by any existing rule

Reply:
Please remove that constraint as we are no longer use that package